src/Transform/OptimizeMemoryPools.cpp

```diff
@@ -29,7 +30,8 @@ using namespace mlir;
 namespace {
 
 // Handling of static memory pool on a block-basis in each function.
-typedef std::map<Block *, bool> BlockToCompactedFlag;
+typedef std::map<Block *, llvm::SmallSet<int64_t, 16>>
```

Comment:
Same thing here, a bit more explanation of what is being tracked. Is it that you are keeping track for each block of the alignment constraints that you have been seen?